### ISSUE TITLE
Parent Request Id is mislabeled as Context Id in log scope

### ISSUE THREAD

**Issue by ngbrown** [aspnet/Hosting#1384]
For the upcoming for ASP.Net Core 2.1; pull #1138 added to the log scope a `ContextId`, but the contents are only the parent request identifier and does not appear to be set if there was no parent request (no `Request-Id` header in the incoming HTTP request).

There is documentation/specifications (mostly by @lmolkova) around these terms at:

* [HttpCorrelationProtocol](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/HttpCorrelationProtocol.md)
* [FlatRequestId](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/FlatRequestId.md)
* [HierarchicalRequestId](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/HierarchicalRequestId.md)

Specifically in the [Activity User Guide](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/ActivityUserGuide.md), it says:

> When application calls external dependency to complete an operation, it may need to pass some of the ___context___ (e.g. correlation id) [emphasis added] along with dependency call to be able to correlate logs from multiple service.

I see that the expectation is that in the logs, there will be a "correlation id" of the context, or `ContextId` if you will, that can be used to link the logs entries across multiple services handling a single user request.

In the current preview code, the Log scope outputs a `ContextId` which is the value of an incoming header `Request-Id`, and only if that header is set.  This is not useful for seeing the entire request across multiple services in the logs.  What is needed is adding to the log scope the root request id to the log scope of all services involved in handling the request, including within the root service that generated the first request id.

**Comment by lmolkova** [comment]
@ngbrown 
The spec implemented in .NET and ASP.NET is [HierarchicalRequestId](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/HierarchicalRequestId.md)

i.e. 'correlationId' looks like `|root-id-that-never-changes.some_unique.suffix.1.2.3.`

You are right that 'parent request id' is the correct name for it. 

Instead of setting parent Request-Id header value to logger scopes, we should use `Activity.Current.Id` that is generated for each request even if there were no Request-Id header.

I believe @davidfowl  hit a snag there because no instrumentation (i.e. Activity creation) happens in absence of ApplicaitonInsights or other tracing systems that lights-up hosting diagnostics.

**Comment by ngbrown** [comment]
If `CorrelationId` in the log entries would most correctly be the `Activity.Current.Id`, then it could be conditionally added, depending on whether or not the `Activity` is created.

If that can't be achieved, then I don't think it should be set with the current name at all.  i.e. change the log scope/enrichment name to `ParentRequestId`.  There are also parameter names that should be updated.

**Comment by davidfowl** [comment]
@lmolkova is right. Maybe we should just remove this from the logging scope altogether.

**Comment by muratg** [comment]
@davidfowl anything to do in 2.2?

**Comment by davidfowl** [comment]
Yes, we should look at this in 2.2 and also look at disabling this by default because of the other security concerns raised here https://github.com/aspnet/Hosting/issues/1385

**Comment by muratg** [comment]
@Tratcher thoughts?

**Comment by Tratcher** [comment]
No specific thoughts, I wasn't involved in most of this work. @pakrym?

**Comment by muratg** [comment]
@davidfowl Moving this out to 3.0.0.
